darts/timeseries.py

```diff
@@ -308,7 +309,25 @@ def from_dataframe(df: pd.DataFrame,
 
         # get time index
         if time_col:
-            time_index = pd.DatetimeIndex(df[time_col])
+            if time_col in df.columns:
+                if np.issubdtype(df[time_col].dtype, object):
+                    try:
+                        time_index = pd.Int64Index(df[time_col].astype(int))
+                    except ValueError:
+                        try:
+                            time_index = pd.DatetimeIndex(df[time_col])
+                        except ValueError:
+                            raise_log(
+                                AttributeError("'time_col' is of 'object' dtype but doesn't contain valid timestamps"))
+                elif np.issubdtype(df[time_col].dtype, np.number):
```

Comment:
Would this work? If yes, I think it's better
```suggestion
                elif np.issubdtype(df[time_col].dtype, np.integer):
```

Reply:
it should according to this: https://numpy.org/doc/stable/reference/arrays.scalars.html#arrays-scalars.